From 5dbfb27c04063132bef568574d1aed54e6c94318 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 29 Jan 2021 19:55:24 +0800
Subject: [PATCH] drm/rockchip: vop2: set config done critical time to lase 1/8
 frame

According to test, 1/4 frame will cost a wait for 4ms, 1/8 frame
cost a wait for 1~2ms, I think 1~2ms is enough for a config done
write.

Change-Id: I6ba46a8a2cb97434998376e85e5b5fc7a53a9616
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 923e25242ddc..0eacbffd3789 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -789,8 +789,8 @@ static inline void vop2_cfg_done(struct drm_crtc *crtc)
 			done_vp = &vop2->vps[vp_id];
 			adjusted_mode = &done_vp->crtc.state->adjusted_mode;
 			vcnt = vop2_read_vcnt(done_vp);
-			/* if close to the last 1/4 frame, wait to next frame */
-			if (vcnt > (adjusted_mode->crtc_vtotal * 3 >> 2)) {
+			/* if close to the last 1/8 frame, wait to next frame */
+			if (vcnt > (adjusted_mode->crtc_vtotal * 7 >> 3)) {
 				vop2_wait_for_fs_by_vcnt(done_vp, vcnt);
 				done_bits = 0;
 			}
-- 
2.35.3

